### ISSUE TITLE
Websocket connection closed Automatically

### ISSUE THREAD

**Issue by Sarankumar-Gembrill** [AIWintermuteAI/WhisperLive#4]
Hey,

I ran the server code using the following command: **python run_server.py --port 9090 --backend faster_whisper**

I checked the server status with: **netstat -ano | findstr :9090**. 

The result was:
**TCP    0.0.0.0:9090           0.0.0.0:0              LISTENING       14748**

Then I tried to connect with the client using example_client.py with this command: **python .\example_client.py**

However, after a few seconds, the WebSocket connection closed automatically. The client terminal showed the following logs:

[INFO]: * recording
[INFO]: Waiting for server ready ...
[INFO]: Opened connection
[INFO]: WebSocket connection closed: 1000:

In the server terminal, I received the following logs:

INFO:websockets.server:connection open
INFO:root:New client connected
ERROR:root:Error during new connection initialization: [WinError 2] The system cannot find the file specified

Can you please help me fix this issue?

**Comment by SoberSalman** [comment]
Same Issue here man, I tried alot of things, like giving it out custom model path, but have no idea what file it is really missing out on

**Comment by SoberSalman** [comment]
I found the Issue, It occurs in the VAD.py where it tries to download the vad model (.onnx )file from a github repo link. Unfortunately that link does not exists anymore so that Model file is never found. I was able to bypass this by placing the file in the onnx folder but still due some model issues. the VAD never gets up and running. If you want to run it, you can run it without VAD at the moment by doing use_vad = False in the example_client.py file. I will hopefully try to fix this one day and make the code up and running for windows

**Comment by jawshi-dev** [comment]
> I found the Issue, It occurs in the VAD.py where it tries to download the vad model (.onnx )file from a github repo link. Unfortunately that link does not exists anymore so that Model file is never found. I was able to bypass this by placing the file in the onnx folder but still due some model issues. the VAD never gets up and running. If you want to run it, you can run it without VAD at the moment by doing use_vad = False in the example_client.py file. I will hopefully try to fix this one day and make the code up and running for windows

Got the same issue. Seems like the guys over at silero-vad moved the file in a different directory. I replaced the URL in the vad.py to the new path, but as you said, it still doesnt work.

**Comment by AIWintermuteAI** [comment]
Hey @jauwshi @SoberSalman @Sarankumar-Gembrill !
Thanks for reporting this. Looks like they updated the model and changed the repository structure.
I fixed it in the latest commit, try pulling the latest master.
You might need to remove the cached silero vad model, it's in your home folder cache.
![Screenshot from 2024-12-18 22-27-13](https://github.com/user-attachments/assets/77d3e048-2f67-4517-937b-b2de37db387a)

**Comment by SoberSalman** [comment]
@AIWintermuteAI It still doesnt work, i did a fresh start and still i dont get that model file in my cache

**Comment by AIWintermuteAI** [comment]
Can you describe what you mean by "fresh start"?

**Comment by SoberSalman** [comment]
I mean I cleaned up my cache files and cloned your repo, but I couldnt find the silero_vad.onnx file in my cache when i ran the server and client. However, i did download the onnx file and placed it where it was looking and it worked now. I ran the wget command you were using in my terminal and it did download that file, but the good thing is IT WORK!!! Really graeful for that. I will see and report if why isnt it downloading the onnx file initally when it is supposed to. You dont have to reopen the issue now
